### PR TITLE
Update to Gradle 7.0.2

### PR DESCRIPTION
**Pull request by A248**
A lot of parts of the build had to be updated to be compatible with Gradle 7. In addition to making the build Gradle 7 compatible, I also replaced use of deprecated methods in Gradle 7 which will be removed in Gradle 8 (which is hopefully a long time from now)

Here's a list of changes:

* `api` and `implementation` replace `compile` when specifying dependencies
* spotless-gradle-plugin, errorprone, the errorprone plugin, and the license-report plugin are all updated
* Maven central is used instead of jcenter
* destinationDir, classifier, and baseName are replaced with the non-deprecated destinationDirectory, archiveClassifier, and archiveBaseName options. I checked to ensure `archiveBaseName.get()` still produces "xtomlj".

This makes the build functional on JDK 16, enhances the dependency metadata for Gradle dependents, and opens up more possibilities for xtomlj in general.

Related to this, would you like to run the Github Actions Gradle workflow on JDK 11 and 16 in addition to 8? I'll make a separate PR for it.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -127,8 +125,8 @@ tasks.withType(JavaCompile) {
   ]
 
   options.errorprone {
-    excludedPaths '.*/generated-src/.*'
```
Comment (kPYKx7Ddw4n1aIKZ) [inline build.gradle:130]: Are you sure that we should remove this line?
This looks like it is excluding ANTLR generated code from static analysis.

I don't think there's much we can do if something is wrong there.

**Review by kPYKx7Ddw4n1aIKZ — CHANGES_REQUESTED** [review]
Looks fine for the most part.

File: build.gradle (hunk 1 above)
Comment (A248) [reply build.gradle:130]: I'll re-add this.

I previously thought `disableWarningsInGeneratedCode` would cover it. However, looking inside the generated code, it seems that `@Generated` is not added by antlr, and this is confirmed by these open issues:
https://github.com/antlr/antlr4/issues/2512
https://github.com/antlr/antlr4/issues/2568

**Comment by A248** [comment]
Fixed a few things - should be ready now.

**Comment by kPYKx7Ddw4n1aIKZ** [comment]
👍

**Pull request merged**